source/installer.py

```diff
@@ -200,20 +200,21 @@ def removeOldProgramFiles(destPath):
 			else:
 				os.remove(fn)
 
-	# #4235: mpr.dll is a Windows system dll accidentally included with
-	# earlier versions of NVDA. Its presence causes problems in Windows Vista.
-	fn = os.path.join(destPath, "mpr.dll")
-	if os.path.isfile(fn):
-		tryRemoveFile(fn)
-
 	# #9960: If compiled python files from older versions aren't removed correctly,
 	# this could cause strange errors when Python tries to create tracebacks
 	# in a newer version of NVDA.
+	#  However don't touch user and system config.
+	#  Also remove old .dll and .manifest files.
 	for curDestDir,subDirs,files in os.walk(destPath):
+		subDirs[:] = [x for x in subDirs if os.path.basename(x).lower() not in (

```

Comment:
This should only be done at the root of destPath. Currently this is being done in every subdirectory.

Reply:
I am afraid I have no idea how to fix this.

Reply:
It should be enough to check whether `curDestDir` is equal to `destPath`.